File: app/src/main/java/com/huhx0015/androidbooster/utils/SnackbarUtils.java
package com.huhx0015.androidbooster.utils;

import com.google.android.material.snackbar.Snackbar;
import android.view.View;
import androidx.annotation.NonNull;

public class SnackbarUtils {

    /** UTILITY METHODS ________________________________________________________________________ **/

    public static void displaySnackbar(@NonNull View parentView, @NonNull String message,
                                       int length, int color) {
        Snackbar snackbar = Snackbar.make(parentView, message, length);
        View snackbarView = snackbar.getView();
        snackbarView.setBackgroundColor(color);
        snackbar.show();
    }

    public static void displaySnackbarWithAction(@NonNull View parentView, @NonNull String message,
                                                 int length, int color, @NonNull String actionText,
                                                 @NonNull View.OnClickListener listener) {
        Snackbar snackbar = Snackbar.make(parentView, message, length)
                .setAction(actionText, listener);
        View snackbarView = snackbar.getView();
        snackbarView.setBackgroundColor(color);
        snackbar.show();
    }
}

File: app/src/main/kotlin/com/huhx0015/androidbooster/utils/SnackbarUtils.kt
package com.huhx0015.androidbooster.utils

import com.google.android.material.snackbar.Snackbar
import android.view.View

object SnackbarUtils {

    /** UTILITY METHODS ________________________________________________________________________  */

    fun displaySnackbar(parentView: View, message: String, length: Int, color: Int) {
        val snackbar = Snackbar.make(parentView, message, length)
        val snackbarView = snackbar.view
        snackbarView.setBackgroundColor(color)
        snackbar.show()
    }

    fun displaySnackbarWithAction(parentView: View, message: String, length: Int, color: Int,
                                  actionText: String, listener: View.OnClickListener) {
        val snackbar = Snackbar.make(parentView, message, length).setAction(actionText, listener)
        val snackbarView = snackbar.view
        snackbarView.setBackgroundColor(color)
        snackbar.show()
    }
}